src/GroupManager.php

```diff
@@ -230,11 +268,85 @@ public function removeGroup($entity_type_id, $bundle_id) {
 
       // Only update and refresh the map if a key was found and unset.
       $editable->set('groups', $groups);
-      $saved = $editable->save();
+      $editable->save();
+
+      // Remove all roles associated with this group type.
+      $this->removeRoles($entity_type_id, $bundle_id);
 
       $this->resetGroupMap();
+    }
+  }
+
+  /**
+   * Creates default roles for the given group type.
```

Comment:
Maybe something like (but that doesn't sound so horrible 😉 ): `Create per bundle roles, based on the default roles.`